amundsen_application/static/js/components/TableDetail/EditableSection/index.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+
+import './styles.scss';
+
+interface EditableSectionProps {
+  title: string;
+}
+
+interface EditableSectionState {
+  isEditing: boolean;
+}
+
+export interface EditableSectionChildProps {
+  isEditing?: boolean;
+  startEditing?: () => void;
+  stopEditing?: () => void;
+}
+
+export class EditableSection extends React.Component<EditableSectionProps, EditableSectionState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isEditing: false,
+    }
+  }
+
+  startEditing = () => {
+    this.setState({ isEditing: true });
+  };
+
+  stopEditing = () => {
+    this.setState({ isEditing: false });
+  };
```

Comment:
Is there a benefit to managing all 2 different methods whose function can be also be achieved with:
```
  setEditMode = (isEditing: boolean) => {
    this.setState({ isEditing });
  };
```

I feel as if this is just as intuitive for the child components to call one method with `true` or `false`, and it is less code to pass around/implement/manage.

Reply:
Originally I tried passing in `toggleEdit = (isEditing: boolean = !this.state.isEditing)` but I couldn't use it as both an `onClick` and a prop to children. I guess I went too far the other direction.

I'll keep `toggleEdit` and `setEditMode` separately.